src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs

```diff
@@ -111,37 +105,34 @@ public OneOrMany<T> RemoveAll(T item)
         {
             if (HasOne)
             {
-                return item.Equals(_one) ? default : this;
+                return EqualityComparer<T>.Default.Equals(item, _one) ? default : this;
             }
 
             var builder = ArrayBuilder<T>.GetInstance();
-            var iter = GetEnumerator();
-            while (iter.MoveNext())
+
+            foreach (var value in _many)
```

Comment:
not really certain why we were hand manipulating Enumerators instead of just foreaching.